src/olympia/versions/compare.py

```diff
@@ -39,15 +43,45 @@ def version_dict(version):
     return vdict
 
 
-def version_int(version):
-    vdict = version_dict(force_text(version))
-    for key in NUMBERS:
-        if not vdict[key]:
-            vdict[key] = 0
+def _get_version_dict(version_string, max_number_minor, max_number_major):
+    vdict = version_dict(
+        force_text(version_string), asterisk_value=max_number_minor,
+        major_asterisk_value=max_number_major)
+    for num in NUMBERS:
+        max_num = max_number_major if num == 'major' else max_number_minor
+        vdict[num] = min(vdict[num] or 0, max_num)
     vdict['alpha'] = {'a': 0, 'b': 1}.get(vdict['alpha'], 2)
     vdict['pre'] = 0 if vdict['pre'] else 1
+    return vdict
+
+
+def version_int(version):
+    """This is used for converting an app version's version string into a
+    single number for comparision.  To maintain compatibility the minor parts
+    are limited to 99 making it unsuitable for comparing addon version strings.
+    """
+    vdict = _get_version_dict(
+        version, max_number_minor=99, max_number_major=MAX_VERSION_PART)
 
     vint = '%d%02d%02d%02d%d%02d%d%02d' % (
         vdict['major'], vdict['minor1'], vdict['minor2'], vdict['minor3'],
         vdict['alpha'], vdict['alpha_ver'], vdict['pre'], vdict['pre_ver'])
-    return min(int(vint), MAXVERSION)
+    return min(int(vint), BIGINT_POSITIVE_MAX)
+
+
+def addon_version_hash(version):
```

Comment:
Not a big fan of calling this an `hash` - it's just a potentially long int... Why not call this `addon_version_int` in fact ?

Reply:
yeah, I went back and forward on returning a string instead of a int

Reply:
https://github.com/mozilla/addons-server/pull/12773/commits/4a7316d59f8020ba0e83a2e1c785c13a36130b9a